sys/cam/scsi/scsi_pass.c

```diff
@@ -1219,12 +1221,14 @@ passcopysglist(struct cam_periph *periph, struct pass_io_req *io_req,
 
 	for (i = 0, j = 0; i < io_req->num_user_segs &&
 	     j < io_req->num_kern_segs;) {
-		uint8_t *user_ptr, *kern_ptr;
+		uint8_t * __capability user_ptr;
+		uint8_t *kern_ptr;
 
 		len_to_copy = min(user_sglist[i].ds_len -user_watermark,
 		    kern_sglist[j].ds_len - kern_watermark);
 
-		user_ptr = (uint8_t *)(uintptr_t)user_sglist[i].ds_addr;
+		user_ptr = __USER_CAP((uint8_t *)(uintptr_t)user_sglist[i].ds_addr,
```

Comment:
Now `__USER_CAP` accepts `uint64_t` without need to cast. I feel it is somewhat redundant as the macro already makes it explicit that the result will be a pointer.
As an aside (and as a note to self), going back to our last conversation about `bus_addr_t`, it seems that there are different interpretations of `bus_addr_t` across architectures on whether it holds a physical address, an opaque value of some kind or a virtual address. This seems to be one of those places that assumes it can hold a virtual address.

Reply:
Here I just kept the existing code, though perhaps I need to adjust this as part of my earlier comment to permit `ds_addr` to be a capability.  I'll know more about how I'd like to handle that (possibly a union around `ds_addr`) when I look at this again.

Reply:
This no longer uses `__USER_CAP`.